cpp/src/io/parquet/page_enc.cu

```diff
@@ -505,11 +506,18 @@ inline __device__ void PackLiterals(
           dst[t * 2 + 1] = v >> 8;
         }
         return;
+      } else if (w == 24) {
+        if (t < count) {
+          dst[t * 3 + 0] = v;
+          dst[t * 3 + 1] = v >> 8;
+          dst[t * 3 + 2] = v >> 16;
+        }
+        return;
       }
     } else {
       return;
     }
-  } else {
+  } else if (w <= 16) {
```

Comment:
How does this else-if get reached over the if clause itself? Are there other values for `w` in between the ones listed in the if-statement `(w == 1 || w == 2 || w == 4 || w == 8 || w == 12 || w == 16 || w == 24)`? Or perhaps I'm reading this wrong.

Reply:
That is correct. For a limited number of values, there's a faster method with a shfl or simply byte copies. The rest need atomics.